sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/implementation/RedirectPolicy.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.communication.common.implementation;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * HttpPipelinePolicy to redirect requests when 302 message is received to the new location marked by the
+ * Location header.
+ */
+public final class RedirectPolicy implements HttpPipelinePolicy {
+    private static final int MAX_REDIRECTS = 10;
```

Comment:
Any concerns around potentially running into a scenario where there are many redirects as well as retries? Since this and the retry policy will both attempt to retry and could result in a multiplicative scenario, ex 3 retry attempts and 10 redirect attempts could result in 30 (possibly 40) network calls before failing.

Reply:
Well no, the policy updates the request when doing the redirection. If there is an error, it is the latest version of the request which will retry.

Reply:
Good point, I also realized that this is short-circuiting redirect location, so having a request amplification isn't as likely.